src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -79,7 +80,14 @@ public CommandResult execute(Model model) throws CommandException {
         Person personToEdit = lastShownList.get(index.getZeroBased());
         Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor);
 
-        if (!personToEdit.isSamePerson(editedPerson) && model.hasPerson(editedPerson)) {
+        // A person's uniqueness is based on their phone and email. If the edit command results in sharing those
+        // identifiers with 2 others, it means the uniqueness invariant was violated and hence there are duplicates.
```

Comment:
I dont think you made it clear why sharing identifiers with 1 other is okay? Sharing 1 mean 2 person is the same right

Maybe it is easer to separate the cases, so it is easier to reason

- personToEdit.isSamePerson == True --> There should only be 1 match which editedPerson will replace, conserving its uniqueness
- personToEdit.isSamePerson == False --> There should be no match and editedPerson is unique 

Just my suggestion tho, you don't have to follow if you can change the comment to be clearer on your approach